Skip and publish performance

robinsio opened the pull request.

jonrobins committed:
production:release - 1.0.16
  - Added consumer skip option
  - Improved publisher performance

jonrobins committed:
production:release - 1.0.16
  - Added consumer skip option
  - Improved publisher performance
  - Updated readme

pa-colin-mchale commented on test/hutch.test.js:
```diff
@@ -280,4 +379,67 @@ describe('Hutch', function() {
+
+        setTimeout(function(){
+          // We should have only processed one of the files.
+          messages[0].should.equal('c2');
```
This checks that the first message processed was by consumer 2, not that only one message was processed.

pa-colin-mchale commented on test/hutch.test.js:
```diff
@@ -280,4 +379,67 @@ describe('Hutch', function() {
+
+  it('should clear skip option for exclusive consumer after consumer is rejected', function(complete) {
+
+    this.timeout(20000); // Allow up to 15 seconds for windows
```
Comment has wrong time.

pa-colin-mchale commented on test/hutch.test.js:
```diff
@@ -83,6 +83,55 @@ describe('Hutch', function() {
+
+        if(count === 2){
+          var channel = hutch._getChannelByExchange(options.exchange.name);
+          channel.should.exist();
```
How does this check it uses the same channel?

robinsio replied:
It doesn't, but it does check to see if we can pull out the channel by exchange which is what happens prior to the publish call

pa-colin-mchale replied:
Maybe change the test description then.

jonrobins committed:
production:release - 1.0.16
  - Added consumer skip option
  - Improved publisher performance
  - Updated readme
  - PR Updates

jonrobins committed:
production:release - 1.0.16
  - Added consumer skip option
  - Improved publisher performance
  - Updated readme
  - PR Updates
  - Better publisher test
